.github/workflows/ci.yml

```diff
@@ -6,55 +6,46 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 2.7]
-        plone-version: [5.2, 5.1, 4.3]
-        exclude:
-          - python-version: 3.7
-            plone-version: 4.3
-          - python-version: 3.7
-            plone-version: 5.1
+        python-version: [2.7, 3.7, 3.8]
+        plone-version: [5.2]
 
     steps:
 
-      # git checkout
       - uses: actions/checkout@v2
 
-      # python setup
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
 
-      # python cache
-      - uses: actions/cache@v1
+      - name: Cache eggs
+        uses: actions/cache@v2
         with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
-
-      # python install
-      - run: pip install virtualenv
-      - run: pip install wheel
-      - name: pip install
-        run: pip install -r requirements.txt
-      - name: choose Plone version
-        run: sed -ie "s#plone-x.x.x.cfg#plone-${{ matrix.plone-version }}.x.cfg#" ci.cfg
-
-      # buildout
+          path: |
+            eggs
+            downloads
+          key: eggs-${{ matrix.python-version }}-${{ hashFiles('**/*.cfg') }}
+          restore-keys: eggs-
+
+      - name: "Install Python dependencies (pip)"
+        uses: "py-actions/py-dependency-install@v2"
+        with:
+          path: "https://dist.plone.org/release/${{ matrix.plone-version }}-latest/requirements.txt"
+          update-setuptools: "false"
+          update-pip: "true"
+          update-wheel: "true"
+
       - name: buildout
-        run: buildout -t 10 -c ci.cfg
+        run: buildout -t 10 -c plone-${{ matrix.plone-version }}.x.cfg code-analysis:return-status-codes=true
```

Comment:
@jensens Thanks for cleaning up! Though, I'd prefer to keep the buildout and CI structure as it is. This package follows the best practices that I use for > 20 packages that I maintain and release on a regular basis. I just can't keep > 20 different buildout/CI configurations up-to-date over time. I am aware that we can and should simplify things in both plone.rest and plone.restapi. Though, I have to find a solution that works for all packages at the same time and not just one. Using different CI settings for the packages I maintain would make it impossible for me to continue to do releases on a regular basis and taking care of the CI setups.

Reply:
That said I would like to see a tox setup in future, as we are planning for other core packages too. But I like to takes this step by step.

Reply:
>  Using different CI settings for the packages I maintain would make it impossible for me to continue to do releases on a regular basis and taking care of the CI setups.

By the way: What about together as a whole community finding a way to maintain all Plone Core packages with need for testing on GH actions? I do not like the idea any maintainer of some core packages invent the wheel and we are ending up with a forest of different approaches. Currently this happens. I do not have found the golden path here, also if it comes to test against releases and the maintained core-dev branches in one GH action run, which would be favorable.

Maybe buildout.plonetest is a good starting point here?

Reply:
@jensens we do not have an official best practice policy for testing individual packages. To lower the maintenance costs I wrote a script that copies over the configuration and that I have to commit manually every single time (which is covered by my signed contributor agreement). I don't think this causes any legal problems and I honestly do not see how that adds a higher barrier. If I would do this manually, nobody would notice any difference at all. Nothing in kitconcept/buildout is company-specific. I use this buildout configuration only for the Plone OS packages that I maintain. I'd be happy to copy that over anywhere if that would make a difference.

Extending buildout.plonetest directly is the wrong approach IMHO because it makes the CI setup rely on an external resource. The build can break without notice and this is an anti-pattern (if not THE anti-pattern) in CI.

The @plone/release-team agreed that I am officially responsible for releasing and maintaining both plone.rest and plone.restapi and that those packages require special handling within Plone core. I can only take over this responsibility when I am able to decide how the CI setup should look like, to make sure the buildout is in a working state all the time and that I can make releases on a regular basis. This is what @esteele, @mauritsvanrees, and I agreed upon.

Reply:
Well, anyway: master CI setup is broken. This fixes it and simplifies everything. If you do want to take this tasks all alone w/o external help your path may be fine. But I know even your time is limited.

Reply:
@jensens I will try to find time to look into it. I also agree that we have to find a way to simplify the buildout configs for our Plone 6 packages.